Implement Error Hooks (#1967)
gruntwork-io/terragrunt: docs/_docs/04_reference/config-blocks-and-attributes.md

- `after_hook` (block): Nested blocks used to specify command hooks that should be run after `terraform` is called.
  Hooks run from the terragrunt configuration directory (the directory where `terragrunt.hcl` lives). Supports the same
  arguments as `before_hook`.

> - `error_hook` (block): Needed blocks used to specify command hooks that run incase of error that matches one of the expression from the list of expressions specified in `on_errors` attribute.

Reviewer: ```suggestion
- `error_hook` (block): Nested blocks used to specify command hooks that run when an error is thrown. The
error must match one of the expressions listed in the `on_errors` attribute.
```

    run_on_error = true
  }

>   # Run incase of any error that occurs during apply or plan

Reviewer: ```suggestion
  # After an error occurs during apply or plan, run "echo Error Hook executed". This hook is configured so that it will run
  # after any error, with the ".*" expression.
```

  error_hook "error_hook_1" {
    commands  = ["apply", "plan"]
    execute   = ["echo", "Error Hook executed"]
    on_errors = [
      ".*",
    ]

>   }

Reviewer: Add an extra line here.

```suggestion
  }
  
```
